expected/16.0/extra/uuid.out

```diff
@@ -362,153 +362,170 @@ SELECT * FROM "type_UUID+";
---Testcase 194:
+--Testcase 094:
 SELECT * FROM "type_UUID+";
  i  |                  u                   |  t   | l  
 ----+--------------------------------------+------+----
- 41 | b0eebc99-9c0b-4ef8-bb6d-6bb900000a15 | blob | 16
+ 41 | b0eebc99-9c0b-4ef8-bb6d-6bb900000a15 | text | 36
  44 |                                      | null |   
 (2 rows)
 
---Testcase 195:
+--Testcase 095:
 SELECT * FROM "type_UUID+" WHERE "u" IS NULL;
  i  | u |  t   | l 
 ----+---+------+---
  44 |   | null |  
 (1 row)
 
---Testcase 196:
+--Testcase 096:
 SELECT * FROM "type_UUID+" WHERE "u" IS NOT NULL;
  i  |                  u                   |  t   | l  
 ----+--------------------------------------+------+----
- 41 | b0eebc99-9c0b-4ef8-bb6d-6bb900000a15 | blob | 16
+ 41 | b0eebc99-9c0b-4ef8-bb6d-6bb900000a15 | text | 36
 (1 row)
 
---Testcase 197:
-EXPLAIN VERBOSE
+--Testcase 097:
+EXPLAIN (VERBOSE, COSTS OFF)
 SELECT * FROM "type_UUID+" WHERE "u" IS NULL;
                                                              QUERY PLAN                                                             
 ------------------------------------------------------------------------------------------------------------------------------------
- Foreign Scan on public."type_UUID+"  (cost=10.00..5.00 rows=5 width=54)
+ Foreign Scan on public."type_UUID+"
    Output: i, u, t, l
    SQLite query: SELECT `i`, sqlite_fdw_uuid_blob(`u`), `t`, `l` FROM main."type_UUID+" WHERE ((sqlite_fdw_uuid_blob(`u`) IS NULL))
 (3 rows)
 
---Testcase 198:
-EXPLAIN VERBOSE
+--Testcase 098:
+EXPLAIN (VERBOSE, COSTS OFF)
 SELECT * FROM "type_UUID+" WHERE "u" IS NOT NULL;
                                                                QUERY PLAN                                                               
 ----------------------------------------------------------------------------------------------------------------------------------------
- Foreign Scan on public."type_UUID+"  (cost=10.00..1045.00 rows=1045 width=54)
+ Foreign Scan on public."type_UUID+"
    Output: i, u, t, l
    SQLite query: SELECT `i`, sqlite_fdw_uuid_blob(`u`), `t`, `l` FROM main."type_UUID+" WHERE ((sqlite_fdw_uuid_blob(`u`) IS NOT NULL))
 (3 rows)
 
---Testcase 47:
+--Testcase 100:
+CREATE FOREIGN TABLE "type_UUIDpk" (col uuid OPTIONS (key 'true')) SERVER sqlite_svr;
+--Testcase 101:
+INSERT INTO "type_UUIDpk" VALUES ('{a0eebc99-9c0b4ef8-bb6d6bb9-bd380a11}');
+--Testcase 102:
+INSERT INTO "type_UUIDpk" VALUES ('{b0eebc99-9c0b4ef8-bb6d6bb9-bd380a12}');
+--Testcase 103: ERR - primary key
+INSERT INTO "type_UUIDpk" VALUES ('{b0eebc99-9c0b4ef8-bb6d6bb9-bd380a12}');
+ERROR:  failed to execute remote SQL: rc=19 UNIQUE constraint failed: type_UUIDpk.col 
+   sql=INSERT INTO main."type_UUIDpk"(`col`) VALUES (?)
+--Testcase 104:
+ALTER FOREIGN TABLE "type_UUIDpk" ALTER COLUMN col OPTIONS (ADD column_type 'BLOB');
+--Testcase 105: NO ERR, but the same semantics!
+INSERT INTO "type_UUIDpk" VALUES ('{b0eebc99-9c0b4ef8-bb6d6bb9-bd380a12}');
```

Comment:
What do you mean `the same semantics`?  
Could you add SELECT test cases after each INSERT test case? It will help developer to verify result after INSERT easily.

Reply:
> What do you mean `the same semantics`?

We can have in SQLite some UUID value as value with `text` affinity and with `blob` affinity. This values will be transformed to equal internal PostgreSQL 16 bytes. This means this values have the same semantics for PostgreSQL, but not for SQLite. SQLite allow unique index with same UUID values with different affinities, but PostgreSQL disallow. This TC is about "SQLite allow unique index with same UUID values".

Additional TCs will be added in the 2nd commit of this review.

Reply:
Thanks for your explanation. I understood.

> Additional TCs will be added in the 2nd commit of this review.

Could you add these test cases?

Reply:
Added in https://github.com/pgspider/sqlite_fdw/pull/95/commits/7cc6fed2240795e80bec225803959a7beb9d1d00 , @son-phamngoc . Please verify. Unfortunately the main change in `sqlite_fdw.c` also was only in private draft, but restored in https://github.com/pgspider/sqlite_fdw/pull/95/commits/7cc6fed2240795e80bec225803959a7beb9d1d00.

Reply:
Note: I am not familiar with `git`, this branch was not synchronized with private draft during missing HEAD commit selecting. Now all is ok after restoring main contribution in https://github.com/pgspider/sqlite_fdw/commit/7cc6fed2240795e80bec225803959a7beb9d1d00 and https://github.com/pgspider/sqlite_fdw/commit/9c81b0fa941c1e56b4c6badcb5820c16c6e5581b

Reply:
Thank you. It's OK now. Please help me to resolve this comment.